synthesizer/src/vm/mod.rs

```diff
@@ -398,7 +398,19 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         self.block_store().pause_atomic_writes()?;
 
         // First, insert the block.
-        self.block_store().insert(block)?;
+        if let Err(insert_error) = self.block_store().insert(block) {
+            if cfg!(feature = "rocks") {
+                // Clear all pending atomic operations so that unpausing the atomic writes
+                // doesn't execute any of the queued storage operations.
+                self.block_store().abort_atomic();
+                // Disable the atomic batch override.
+                // Note: This call is guaranteed to succeed (without error), because `DISCARD_BATCH == true`.
+                self.block_store().unpause_atomic_writes::<true>()?;
```

Comment:
This always should have been there; block insertion is the only operation where pausing is used (because block insertion and finalization are technically separate atomic operations, but they need to be treated atomically here), so once the pause is in force, it MUST be unpaused by the end of that operation or if that operation fails (which is covered by this PR).